docs/website/root/mithril/threat-model.md

```diff
@@ -0,0 +1,318 @@
+
+Mithril clients do connect to an aggregator using HTTP over TLS to query Mithril certificates for certified chain data and locate artifacts.
+
+A Mithril client can verify the received Mithril certificate is linked to other certificates up to the genesis certificate and can be verified using the Mithril genesis verification key (see [details](https://mithril.network/doc/mithril/mithril-protocol/certificates/)).
+
+:::info To do
+
+Missing: the currently recommended relay (reverse proxy)
+
+:::
+
+#### Deployment architecture
+
+This document is specifically targeting the standard deployment architecture where a Mithril signer runs next to the block producing node, while access to the Mithril aggregator is only done through a relay.
+
+[![Mithril - Container View](./images/deployment-architecture.jpg)](./images/deployment-architecture.jpg)
+
+### External Dependencies
+
+Listing dependencies for mithril-signer
+```
+% cargo-deps-list -e normal -p mithril-signer 
+...
+slog-json v2.6.1 {default}
+slog-scope v4.4.0
+arc-swap v1.6.0
+
+Total dependencies: 267
+```
+
+
+
+<!-- * [libp2p](https://docs.libp2p.io/)
+    * https://docs.libp2p.io/concepts/security/security-considerations/
+    * [DoS Mitigation](https://docs.libp2p.io/concepts/security/dos-mitigation/) -->
+    
+* SPOs infrastructure:
+    * Block producing host configuration
+    * Relay hosts configuration
+    * Firewall/private network
+
+### Entry Points 
+
+* mithril-relay P2P TCP ports
+* mithril-aggregator HTTP port
+* mithril-relay HTTP port
+
+## Assumptions 
+
+The core Mithril protocol is considered safe and its analysis is out of scope for this document, it is computationally infeasible to:
+* Forge a valid aggregate signature from forged signing keys
+* Forge individual signatures impersonating one of the signers
+
+## Assets
+
+For each asset we first identify what protection is required: Availability, Confidentiality, Integrity ie. the [CIA Triad](https://www.splunk.com/en_us/blog/learn/cia-triad-confidentiality-integrity-availability.html)
+
+:::info Note
+
+I follow Pascal's suggestion to identify assets as processes rather than particular resources, except in the case of actual pieces of data (eg. keys). The latter should be listed in the threats section and their protection mitigated
+
+:::
+
+We then identify threats and countermeasures
+
+#### KES private keys
+
+* KES key is present only on BP Node but needs to be shared with both the cardano-node process and the mithril-signer process
+* KES keys are needed by mithril-signer in order to sign verification key along with an operational certificate which authenticates the key from this stake pool id
+* This signing happens at every epoch
+    * **availability**: Yes (If KES key is unavailable then signing cannot proceeed)
+    * **confidentiality**: Yes (Capturing KES private keys allow an attacker to impersonate a registered SPO on-chain and produce blocks on his behalf)
```

Comment:
Should we distinguish between the current KES key from the cold keys? AFAIK does breach on confidentiality of the current KES key allow to impersonate an SPO **until rotated**. That would limit the risk temporally

Reply:
As we only use KES keys and Operational Certificate, I guess that it is not needed to mention the Cold keys. 
However, we should probably add a sub section for the Operational Certificate, WDYT?

Also, you're right as the impersonation of an SPO is possible only until the KES keys are rotated, I updated the sentence in that sense :+1: